public/components/jobHistoryDateRange.react.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import R from 'ramda';
+
+
+export default class JobHistoryDateRange extends React.Component {
+
+    constructor(props) {
+        super(props);
+    }
+
+    empty(value) {
+        return R.isNil(value) || R.isEmpty(value);
+    }
+
+    nonEmpty(value) {
+        return !this.empty(value);
+    }
+
+    render () {
+        if (this.empty(this.props.rangeFrom) && this.empty(this.props.rangeTo)) {
+            return <span>All time</span>;
+        } else if (this.nonEmpty(this.props.rangeFrom) && this.empty(this.props.rangeTo)) {
+            return <span>{ new Date(this.props.rangeFrom).toISOString().slice(0,10) + " - now" }</span>;
```

Comment:
This has no benefit but you might find moment.js preferable to use. See: https://github.com/guardian/floodgate/blob/ef2330020d3ab640101b580483c2493adc5b40f4/public/components/reindexForm.react.js#L12